src/software/ai/passing/chip_pass.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "software/ai/passing/base_pass.h"
+#include "shared/constants.h"
+
+class ChipPass: public BasePass
+{
+    public:
+    ChipPass() = delete;
+
+    /**
+     * Creates a chip pass from the given destination point
+     *
+     * @param passer_point the starting point of the pass
+     * @param pass_destination the end point of the pass
+     * @return the ChipPass constructed from the start and end points
+     */
+    ChipPass(Point passer_point, Point receiver_point);
+
+    double firstBounceRange();
+
+    virtual Duration estimatePassDuration() const;
```

Comment:
you should use `override` so that it checks at compile time that you're overriding the base class implementation corrctly